apps/aehttp/src/sc_ws_api_jsonrpc.erl

```diff
@@ -349,6 +349,24 @@ process_request(#{<<"method">> := <<"channels.get.offchain_state">>,
                     , {int, type} => reply
                     , payload => aesc_offchain_state:serialize_for_client(State) }}
     end;
+process_request(#{<<"method">> := <<"channels.get.contract">>,
+                  <<"params">> := #{<<"pubkey">> := PubkeyE}}, FsmPid) ->
+    case aehttp_api_encoder:safe_decode(contract_pubkey, PubkeyE) of
+        {ok, Pubkey} ->
+            case aesc_fsm:get_contract(FsmPid, Pubkey) of
+                {ok, Contract} ->
+                    ContractState = aect_contracts:state(Contract),
+                    Resp = #{<<"contract">> => aect_contracts:serialize_for_client(Contract),
```

Comment:
I see the contract's serialization [does not include](https://github.com/aeternity/aeternity/blob/master/apps/aecontract/src/aect_contracts.erl#L204_L220) the state, hence the following line. Ok

Reply:
Yes, this makes sense as the contract state might be big. Here we specifically want to have it, but in general one might want only other parts of channel. Also it's stored separately.